src/t8_cmesh/t8_cmesh_readmshfile.cxx

```diff
@@ -832,14 +838,30 @@ t8_cmesh_msh_file_4_read_eles (t8_cmesh_t cmesh, FILE *fp,
       }
     }
     else {
-      for (tree_loop = 0; tree_loop < num_ele_in_block; tree_loop++) {
+      for (tree_loop = 0; tree_loop < num_ele_in_block;
+           tree_loop++, tree_count++) {
         /* Read the next line containing tree information */
         retval = t8_cmesh_msh_read_next_line (&line, &linen, fp);
         if (retval < 0) {
           t8_global_errorf ("Premature end of line while reading trees.\n");
           goto die_ele;
         }
         t8_cmesh_set_tree_class (cmesh, tree_count, eclass);
+
+        /* Set the geometry of the tree to be occ or linear */
+        /* TODO: We should optimize here and only set the geometry to be 
+         *        occ if the tree actually has any curved data. */
```

Comment:
Yes, all trees in the "free flow" of a 3D mesh are linear, because they were not generated on a CAD geometry. There is the follow-up PR [#417](https://github.com/DLR-AMR/t8code/pull/417), which takes this into account, but it is only mergeable, if this PR gets merged.